plugin/src/main/java/org/opensearch/ml/action/model_group/GetModelGroupTransportAction.java

```diff
@@ -42,83 +47,124 @@
         this.modelAccessControlHelper = modelAccessControlHelper;
+        this.mlFeatureEnabledSetting = mlFeatureEnabledSetting;
     }
 
     @Override
     protected void doExecute(Task task, ActionRequest request, ActionListener<MLModelGroupGetResponse> actionListener) {
         MLModelGroupGetRequest mlModelGroupGetRequest = MLModelGroupGetRequest.fromActionRequest(request);
         String modelGroupId = mlModelGroupGetRequest.getModelGroupId();
-        GetRequest getRequest = new GetRequest(ML_MODEL_GROUP_INDEX).id(modelGroupId);
+        String tenantId = mlModelGroupGetRequest.getTenantId();
+        if (!TenantAwareHelper.validateTenantId(mlFeatureEnabledSetting, tenantId, actionListener)) {
+            return;
+        }
+
+        FetchSourceContext fetchSourceContext = new FetchSourceContext(true, Strings.EMPTY_ARRAY, Strings.EMPTY_ARRAY);
+        GetDataObjectRequest getDataObjectRequest = new GetDataObjectRequest.Builder()
+            .index(ML_MODEL_GROUP_INDEX)
+            .id(modelGroupId)
+            .tenantId(tenantId)
+            .fetchSourceContext(fetchSourceContext)
+            .build();
+
         User user = RestActionUtils.getUserContext(client);
 
         try (ThreadContext.StoredContext context = client.threadPool().getThreadContext().stashContext()) {
-            ActionListener<MLModelGroupGetResponse> wrappedListener = ActionListener.runBefore(actionListener, () -> context.restore());
-            client.get(getRequest, ActionListener.wrap(r -> {
-                if (r != null && r.isExists()) {
-                    try (XContentParser parser = createXContentParserFromRegistry(xContentRegistry, r.getSourceAsBytesRef())) {
-                        ensureExpectedToken(XContentParser.Token.START_OBJECT, parser.nextToken(), parser);
+            ActionListener<MLModelGroupGetResponse> wrappedListener = ActionListener.runBefore(actionListener, context::restore);
 
-                        MLModelGroup mlModelGroup = MLModelGroup.parse(parser);
-                        modelAccessControlHelper.validateModelGroupAccess(user, modelGroupId, client, ActionListener.wrap(access -> {
-                            if (!access) {
-                                wrappedListener
-                                    .onFailure(
-                                        new OpenSearchStatusException(
-                                            "User doesn't have privilege to perform this operation on this model group",
-                                            RestStatus.FORBIDDEN
-                                        )
-                                    );
+            sdkClient
+                .getDataObjectAsync(getDataObjectRequest, client.threadPool().executor(GENERAL_THREAD_POOL))
+                .whenComplete((r, throwable) -> {
+                    log.debug("Completed Get Model group Request, id:{}", modelGroupId);
+                    if (throwable != null) {
```

Comment:
Generally it is better to handle exception using try catch blocks. Can we follow a pattern similar to https://github.com/opensearch-project/ml-commons/blob/c15ca42385312a92e1f0e4829f01f4b1b31c1008/common/src/main/java/org/opensearch/sdk/SdkClient.java#L76-L79

Reply:
@arjunkumargiri see my reply to the high level comment earlier.

If we were writing code from scratch I'd be completely on board with this pattern.   New code probably _should_ follow this pattern.

But we're not writing new code. We're migrating existing code that is designed to handle a `void` return method and deal with response or exception in an `ActionListener` and using this `throwable == null` or `response == null` (or their opposites) pattern minimizes the effort involved in migrating and speeds the process.